src/right-field/RightField.jsx

```diff
@@ -130,49 +172,118 @@ function bindingStreamAdder(item, variables, setter) {
- * @param {Function} setIsQuerying boolean setter function for whether the worker is still executing a query or not
+ * If there is currently an iterator active iterator, either a BindingStream or QuadStream, it removes all event listeners and stops the execution.
  */
-function configureQueryWorker(adder, variableSetter, setIsQuerying) {
-  queryWorker = new QueryWorker();
-  let variablesMain = [];
-  queryWorker.onmessage = ({ data }) => {
-    switch (data.type) {
-      case "result":
-        let binding = JSON.parse(data.result);
-        let entries = binding.entries;
-        adder(entries, variablesMain);
-
-        break;
-      case "end":
-        setIsQuerying(false);
-        break;
-      case "metadata": {
-        variablesMain = data.metadata.variables.map((val) => val.value);
-        variableSetter(variablesMain);
-        break;
-      }
-    }
-  };
+function disableIterator() {
+  if (iterator) {
+    iterator.removeAllListeners("end");
+    iterator.removeAllListeners("data");
+    iterator.removeAllListeners("error");
+    iterator.destroy();
+  }
 }
 
 /**
- * A function that executes a given query and processes every result as a stream based on the functions provided.
- * @param {query} query the query which gets executed
- * @param {Worker} queryWorker the worker which will be used to execute the given query
+ * A function that executes a given query and processes every result as a stream based on the EventEmitter.
+ * @param {query} query the query which is to be executed
+ * @param {EventEmitter} eventEmitter an EventEmitter that listens to and emits UI state changes.
  */
-async function executeQuery(query, queryWorker) {
+async function executeQuery(query, eventEmitter) {
   try {
     let result = await fetch(`${config.queryFolder}${query.queryLocation}`);
     query.queryText = await result.text();
-    queryWorker.postMessage({ selectedQuery: query });
+    let fetchFunction = getDefaultSession().info.isLoggedIn ? authFetch : fetch;
+    return handleQueryExecution(
+      await myEngine.query(query.queryText, {
+        sources: query.sources,
+        fetch: fetchFunction,
+      }),
+      eventEmitter
+    );
   } catch (error) {
+    eventEmitter.emit("queryingStatus", false);
     handleQueryFetchFail(error);
   }
 }
 
+/**
+ * A function that given a QueryType processes every result as a stream.
+ *
+ * @param {QueryType} execution a query execution
+ * @param {EventEmitter} eventEmitter an EventEmitter that listens to and emits UI state changes.
+ */
+async function handleQueryExecution(execution, eventEmitter) {
+  try {
+    let metadata = await execution.metadata();
+    let variables = metadata.variables.map((val) => {
+      return val.value;
+    });
+
+    eventEmitter.emit("variables", variables);
+
+    queryTypeHandlers[execution.resultType](
+      await execution.execute(),
+      variables,
+      eventEmitter
+    );
+  } catch (error) {
+    console.error(error.message); //TODO
```

Comment:
There should not be any todos left.

Reply:
This TODO is handled in PR #24

Reply:
I would suggest doing it as follows next time: create an issue for the todo and mention the link of the issue in the code.